addons/web/static/src/views/graph/graph_renderer.js

```diff
@@ -15,7 +15,107 @@ import { ReportViewMeasures } from "@web/views/view_components/report_view_measu
+const GRAPH_LABEL_COLOR = getCustomColor(colorScheme, "#111827", "#E4E4E4");
+
+/**
+ * Custom Plugin for Line chart:
+ * Draw the scale grid on top of the chart to
+ * see this last one correctly.
+ */
+const gridOnTop = {
+    id: 'gridOnTop',
+    afterDraw: (chart, args, options) => {
+        if(chart.config.options.plugins.gridOnTop) {
+            const elements = chart.getDatasetMeta(0).data || [];
+            const ctx = chart.ctx;
+            const chartArea = chart.chartArea;
+            const yAxis = chart.scales.y;
+            const xAxis = chart.scales.x;
+
+            ctx.lineWidth = options.gridLineWidth;
+            ctx.strokeStyle = options.gridColor;
+
+            // Draw Y axis scale
+            yAxis.ticks.forEach((value, index) => {
+                const y = yAxis.getPixelForTick(index);
+                ctx.beginPath();
+                // Draw the line scale
+                ctx.moveTo(chartArea.left, y);
+                ctx.lineTo(chartArea.right, y);
+                // Draw the tick mark
+                ctx.moveTo(chartArea.left - options.tickLength, y);
+                ctx.lineTo(chartArea.left, y);
+                ctx.setLineDash([]);
+                ctx.stroke();
+            });
+
+            // Draw X axis tick marks
+            xAxis.ticks.forEach((value, tickIndex) => {
+                const x = xAxis.getPixelForTick(tickIndex);
+                ctx.beginPath();
+                ctx.moveTo(x, chartArea.bottom);
+                ctx.lineTo(x, chartArea.bottom + options.tickLength);
+                ctx.stroke();
+            });
+
+            // Draw the X axis dashed line
+            elements.forEach((point, eltIndex) => {
+                xAxis.ticks.forEach((value, tickIndex) => {
+                    if(point.active && eltIndex === tickIndex) {
+                        const x = xAxis.getPixelForTick(tickIndex);
+                        ctx.beginPath();
+                        ctx.moveTo(x, chartArea.top,);
+                        ctx.lineTo(x, chartArea.bottom);
+                        ctx.strokeStyle = options.highlightedGridColor;
+                        ctx.stroke();
+                    }
+                });
+            });
+        }
+    }
+};
+
+/**
+ * Used to generate the max value of the grid (line & bar charts).
+ * The purpose is to keep a bit of space between the highest data
+ * and the top of the grid.
+ * @param {Array} data
+ * @param {Boolean} isStacked
+ * @returns {Number}
+ */
+function getMaxValue(data, isStacked) {
+    let max = 0;
+    // If there is data
+    if(data.datasets.length) {
+        if(isStacked) {
+            // Initialize an array to define the addition of values according to their index
+            let maxValues = [];
+            data.datasets[0].data.forEach((_, index) => {
+                maxValues[index] = 0;
+            });
+            // Add values for each index
+            data.datasets.forEach(dataset => {
```

Comment:
I think dataset have not always the same length for their data. I think you should avoid the initialization above
and instead have
maxValues[index] = (maxValues[index] || 0)  + value;